src/Compilers/CSharp/Test/Symbol/SymbolDisplay/SymbolDisplayTests.cs

```diff
@@ -7652,5 +7652,140 @@ public void RecordDeclaration()
                 SymbolDisplayPartKind.Space,
                 SymbolDisplayPartKind.RecordClassName);
         }
+
+        [Fact]
+        [WorkItem(48023, "https://github.com/dotnet/roslyn/issues/48023")]
+        public void NullableAttributes_MayBeNullOnClassProperty()
+        {
+            var source =
+@"
+using System.Diagnostics.CodeAnalysis;
+
+#nullable enable
+
+public class A<T>
+{
+    [MaybeNull]
+    public T P { get; }
+}
+
+public class B
+{
+    void M()
+    {
+        var a = new A<object>();
+        _ = a.P;
+    }
+}
+";
+            var compilation = CreateCompilation(new[] { source, MaybeNullAttributeDefinition });
+            var tree = compilation.SyntaxTrees[0];
+            var model = compilation.GetSemanticModel(tree);
+
+            // look for memberAccess "a.P"
+            var memberAccess = tree.GetRoot().DescendantNodes().OfType<AssignmentExpressionSyntax>().Single().Right;
+            var symbol = model.GetSymbolInfo(memberAccess).Symbol;
```

Comment:
>var symbol = model.GetSymbolInfo(memberAccess).Symbol; [](start = 12, length = 54)

It is an overkill to use syntax and SemanticModel to get to a symbol for a type memeber. Symbols can be accessed directly from a compilation. We also have helpers like ```GetMember```, etc.

Reply:
That pattern is often used in that test class and `GetMember` would return `P` of the open generic type, while I need the closed generic type.

Reply:
> That pattern is often used in that test class and GetMember would return P of the open generic type, while I need the closed generic type.

I explained how to get symbol for the closed type below (for NullableAttributes_MayBeNullOnStructProperty unit-test). We also need to test behavior of the API for an opened type. Another thing to consider is that C# SemanticModel is not going to help with getting VB symbols, whereas getting them through symbol model is going to be pretty much the same.

---
In reply to: [558518218](https://github.com/dotnet/roslyn/pull/50527#discussion_r558518218) [](ancestors = 558518218)

Reply:
Done in 71b3eac7351c87d232db82d3a8ed5c62b1f1e29e

Reply:
> We also need to test behavior of the API for an opened type

Added verification for open types in ddf3d061 and 13c5cf39